include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -336,30 +336,31 @@ __parallel_transform_reduce(_ExecutionPolicy&& __exec, _Up __u, _Cp __combine, _
     __wgroup_size = ::std::min(__wgroup_size, oneapi::dpl::__internal::__kernel_work_group_size(
                                                   ::std::forward<_ExecutionPolicy>(__exec), __kernel));
 #endif
-    auto __mcu = oneapi::dpl::__internal::__max_compute_units(::std::forward<_ExecutionPolicy>(__exec));
-    auto __n_groups = (__n - 1) / __wgroup_size + 1;
-    __n_groups = ::std::min(decltype(__n_groups)(__mcu), __n_groups);
-    // TODO: try to change __n_groups with another formula for more perfect load balancing
+    // It can be replaced with another value for more perfect load balancing
+    auto __iters_per_witem = decltype(__wgroup_size)(4);
+    auto __size_per_wg = __iters_per_witem * __wgroup_size; // how much buffer elements work group processes
+    auto __n_groups = (__n - 1) / __size_per_wg + 1;        // number of work groups
+    auto __n_blocks = (__n - 1) / __iters_per_witem + 1;    // number of bunch of elements that the algorithm processes
 
-    _PRINT_INFO_IN_DEBUG_MODE(__exec, __wgroup_size, __mcu);
+    _PRINT_INFO_IN_DEBUG_MODE(__exec, __wgroup_size);
 
     // Create temporary global buffers to store temporary values
-    auto __temp_1 = sycl::buffer<_Tp>(sycl::range<1>(__n_groups));
-    auto __temp_2 = sycl::buffer<_Tp>(sycl::range<1>(__n_groups));
+    auto __temp_1 = sycl::buffer<_Tp>(sycl::range<1>(2 * __n_groups));
     // __is_first == true. Reduce over each work_group
     // __is_first == false. Reduce between work groups
     bool __is_first = true;
-    auto __buf_1_ptr = &__temp_1; // __buf_1_ptr is not accessed on the device when __is_first == true
-    auto __buf_2_ptr = &__temp_1;
+
+    auto __offset_1 = decltype(__n_groups)(0);
```

Comment:
Maybe something like that?
`// For memory utilization it's better to use one bigger buffer instead of two smaller because size of the buffer is close to a few MB`